c-dependencies/js-compute-runtime/allocator.cpp

```diff
@@ -0,0 +1,17 @@
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-offsetof"
+#pragma clang diagnostic ignored "-Wdeprecated-enum-enum-conversion"
+#include "jsapi.h"
+#pragma clang diagnostic pop
+
+JSContext *CONTEXT = nullptr;
+
+extern "C" {
+
+void *cabi_realloc(void *ptr, size_t orig_size, size_t align, size_t new_size) {
```

Comment:
Should this be:

```suggestion
__attribute__((weak, export_name("cabi_realloc")))
void *cabi_realloc(void *ptr, size_t orig_size, size_t align, size_t new_size) {
```

Note that it is a `weak` symbol because the separate bindgen file from wit bindgen also defines its own alloc function which is being overridden.

Reply:
I've moved it there, but am intentionally not making it `weak` as I'd like to ensure that the one generated by `wit-bindgen` is overridden by this one during linking.